lib/std/builtin.zig

```diff
@@ -141,23 +141,49 @@ pub const Mode = enum {
 /// This data structure is used by the Zig language code generation and
 /// therefore must be kept in sync with the compiler implementation.
 pub const CallingConvention = enum {
+    /// This is the default Zig calling convention used when not using `export` on `fn`
+    /// and no other calling convention is specified.
     Unspecified,
+    /// Matches the C ABI for the target.
```

Comment:
That is not too precise, because there are multiple calling conventions involved (arch, platform, libc) but probably close enough.

Reply:
https://en.wikipedia.org/wiki/Application_binary_interface#Description
arch: Processor instruction set, Sizes, layouts, and alignments of basic data types.
c+arch: https://en.wikipedia.org/wiki/Calling_convention
kernel: system calls

AFAIK what is understood as C abi is not a "complete abi", which would require stuff like symbol versioning (libcs like glibc had breaking changes and those are not specified and can be looked up machine readable at call time).

Reply:
Well it's probably close enough like you said. Here's a place in the langref where I found it documented this way too:
```
// The export specifier makes a function externally visible in the generated
// object file, and makes it use the C ABI.
export fn sub(a: i8, b: i8) i8 { return a - b; }
```